__tests__/controllers/users.controller.spec.ts

```diff
@@ -0,0 +1,64 @@
+import * as Faker from 'faker';
+import request from 'supertest';
+import connection from '../../src/database/connection';
+import { getRepository } from 'typeorm';
+import { User } from '../../src/entities/user.entity';
+import app from '../../src/app';
+
+const api = '/api/v1';
+
+beforeAll(async () => {
+  await connection.create();
+});
+
+afterAll(async () => {
+  await connection.close();
+});
+
+beforeEach(async () => {
+  await connection.clear();
+});
+
+describe('when requesting all users', () => {
+  it('should return 500 without authentication', async () => {
+    const response = await request(app).get(`${api}/users`);
+    expect(response.status).toBe(500);
+  });
+});
+
+describe('when requesting one user', () => {
+  it('should return 404 for non-existing user', async () => {
+    const random = Math.round(Math.random() * 10);
+    const response = await request(app).get(`${api}/users/${random}`);
+    expect(response.status).toBe(404);
+  });
+});
+
+describe('when creating a user', () => {
+  it('should return 200 and create db record', async () => {
+    const userFields = {
```

Comment:
Not sure but maybe we can do something like `const user = await factory(User)().create()`.

Reply:
yeah, I tried to use it but didn't know how to integrate it with supertest so it works as real requests.